cg/services/order_validation_service/rules/sample/rules.py

```diff
@@ -0,0 +1,341 @@
+) -> list[OrganismDoesNotExistError]:
+    """
+    Validate that the organisms of all samples in the order exist in the database.
+    Only applicable to order types with non-human samples.
+    """
+    errors: list[OrganismDoesNotExistError] = []
+    for sample_index, sample in order.enumerated_samples:
+        if not store.get_organism_by_internal_id(sample.organism):
+            error = OrganismDoesNotExistError(sample_index=sample_index)
+            errors.append(error)
+    return errors
+
+
+def validate_pools_contain_one_application(
+    order: OrderWithIndexedSamples, **kwargs
+) -> list[PoolApplicationError]:
+    """
+    Validate that the pools in the order contain only samples with the same application.
+    Only applicable to order types with indexed samples (RML and Fluffy).
+    """
+    errors: list[PoolApplicationError] = []
+    for pool, enumerated_samples in order.enumerated_pools.items():
+        samples: list[IndexedSample] = [sample for _, sample in enumerated_samples]
+        if has_multiple_applications(samples):
+            for sample_index, _ in enumerated_samples:
+                error = PoolApplicationError(sample_index=sample_index, pool_name=pool)
+                errors.append(error)
+    return errors
+
+
+def validate_pools_contain_one_priority(
+    order: OrderWithIndexedSamples, **kwargs
+) -> list[PoolPriorityError]:
+    """
+    Validate that the pools in the order contain only samples with the same priority.
+    Only applicable to order types with indexed samples (RML and Fluffy).
+    """
+    errors: list[PoolPriorityError] = []
+    for pool, enumerated_samples in order.enumerated_pools.items():
+        samples: list[IndexedSample] = [sample for _, sample in enumerated_samples]
+        if has_multiple_priorities(samples):
+            for sample_index, _ in enumerated_samples:
+                error = PoolPriorityError(sample_index=sample_index, pool_name=pool)
+                errors.append(error)
+    return errors
+
+
+def validate_sample_names_available(
+    order: OrderWithSamples, store: Store, **kwargs
+) -> list[SampleNameNotAvailableError]:
+    """
+    Validate that the sample names do not exists in the database under the same customer.
+    Applicable to all order types.
+    """
+    errors: list[SampleNameNotAvailableError] = []
+    customer = store.get_customer_by_internal_id(order.customer)
+    for sample_index, sample in order.enumerated_samples:
+        if store.get_sample_by_customer_and_name(
+            sample_name=sample.name, customer_entry_id=[customer.id]
+        ):
+            error = SampleNameNotAvailableError(sample_index=sample_index)
+            errors.append(error)
+    return errors
+
+
+def validate_sample_names_unique(
+    order: OrderWithSamples, **kwargs
+) -> list[SampleNameRepeatedError]:
+    """
+    Validate that all the sample names are unique within the order.
+    Applicable to all order types except Mutant orders.
```

Comment:
Implement this in mutant https://github.com/Clinical-Genomics/improve-order-flow/issues/102

Reply:
not applicable, resolving